### PR TITLE
[vm] Invalidate loader cache on module loading errors

### PR DESCRIPTION
**Pull request by wrwg**
### Description

This invalidates the loader cache if an error happens either in module bundle loading or execution of `init_module`. In such case, none of the modules is committed to storage, but some may linger in the cache. This can lead to inconsistencies, like binding to `init_module` of an older (failed) version of a module.

### Test Plan

The added test fails with the code before this PR, and succeeds with this one.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4668)

### CODE REVIEW

**Review by davidiw — APPROVED** [review]

**Review by zekun000 — APPROVED** [review]

File: aptos-move/framework/cached-packages/src/aptos_token_sdk_builder.rs [hunk 1]
```diff
@@ -253,22 +253,22 @@ impl EntryFunctionCall {
...
             TokenCoinSwapListTokenForSwap {
                 coin_type,
-                creators_address,
-                collection,
-                name,
-                property_version,
-                token_amount,
-                min_coin_per_token,
-                locked_until_secs,
+                _creators_address,
```
Comment (runtian-zhou) [inline aptos-move/framework/cached-packages/src/aptos_token_sdk_builder.rs:256]: why is this file changed?

**Review by runtian-zhou — APPROVED** [review]

File: aptos-move/framework/cached-packages/src/aptos_token_sdk_builder.rs (hunk 1 above)
Comment (wrwg) [reply aptos-move/framework/cached-packages/src/aptos_token_sdk_builder.rs:256]: Known problem, unrelated.
